ister.py

```diff
@@ -1434,21 +1460,23 @@ def handle_options():
     group.add_argument("-F", "--fast-install", action="store_true",
                         default=False,
                         help="Move swupd state dir inside image for a faster install")
-    args = parser.parse_args()
+    args = parser.parse_args(sys_args)
     return args
 
 
 def main():
     """Start the installer
     """
     global LOG
-    args = handle_options()
+    args = handle_options(sys.argv[1:])
 
     LOG = logging.getLogger(__name__)
     handle_logging(args.loglevel, args.logfile)
 
     try:
-        install_os(args)
+        configuration = parse_config(args)
+        template = get_template(configuration["template"])
```

Comment:
Since we get the template here it would make sense to me to call validate_template right after instead of inside the install_os function.

Reply:
if we use template inside install_os we will force the creation of the template in the hard disk when ister is used by iste_gui, by this way we only provide a template if we run this in the command line.

Reply:
Okay, yea I'd say from an ister as API perspective install_os is where we expect folks to call into and we don't expect them to validate the template they create as we won't want to just run off with that. Agreed.